nilearn/input_data/nifti_labels_masker.py

```diff
@@ -178,6 +186,14 @@ def __init__(self, labels_img, background_label=0, mask_img=None,
             raise ValueError("invalid value for 'resampling_target' "
                              "parameter: " + str(resampling_target))
 
+        if float(n_jobs) < -1 or not float(n_jobs).is_integer() or \
```

Comment:
these checks look like they should either happen for all maskers or none of them

Reply:
@jeromedockes -- I've gone ahead and relocated the checks. Please let me know if the most recent push doesn't address your concern here, in which case let me know how you'd prefer this be handled.